File: src/main/java/net/frozenblock/configurableeverything/config/GameConfig.java
package net.frozenblock.configurableeverything.config;

import blue.endless.jankson.Comment;
import net.frozenblock.configurableeverything.util.ConfigurableEverythingSharedConstantsKt;
import net.frozenblock.configurableeverything.util.ConfigurableEverythingUtilsKt;
import net.frozenblock.lib.config.api.instance.Config;
import net.frozenblock.lib.config.api.instance.json.JsonConfig;
import net.frozenblock.lib.config.api.registry.ConfigRegistry;

public class GameConfig {

	private static final Config<GameConfig> INSTANCE = ConfigRegistry.register(
		new JsonConfig<>(
			ConfigurableEverythingSharedConstantsKt.MOD_ID,
			GameConfig.class,
			ConfigurableEverythingUtilsKt.makeConfigPath("game", true),
			true
		)
	);

	@Comment("Changes the game window title. Change this from an empty string to apply.")
	public String windowTitle = "";

	@Comment(
		"""
		Changes the version series. Change this from an empty string to apply.
		The default version series is "main".
		"""
	)
	public String versionSeries = "";

	public static GameConfig get() {
		return INSTANCE.config();
	}
}

File: src/main/java/net/frozenblock/configurableeverything/config/GameConfig.kt
package net.frozenblock.configurableeverything.config

import net.frozenblock.configurableeverything.util.CEConfig
import net.frozenblock.configurableeverything.util.CONFIG_FORMAT
import net.frozenblock.configurableeverything.util.MOD_ID
import net.frozenblock.configurableeverything.util.makeConfigPath
import net.frozenblock.lib.config.api.instance.xjs.XjsConfig
import net.frozenblock.lib.config.api.registry.ConfigRegistry
import net.frozenblock.lib.config.api.sync.SyncBehavior
import net.frozenblock.lib.config.api.sync.annotation.EntrySyncData
import net.frozenblock.lib.config.api.sync.annotation.UnsyncableConfig
import blue.endless.jankson.Comment

@UnsyncableConfig
data class GameConfig(
    @JvmField
    @EntrySyncData(behavior = SyncBehavior.UNSYNCABLE)
    @Comment("Changes the game window title. Change this from an empty string to apply.")
    var windowTitle: String = "",

    @JvmField
    @EntrySyncData(behavior = SyncBehavior.UNSYNCABLE)
    @Comment(
"""
Changes the version series. Change this from an empty string to apply.
The default version series is "main".
"""
    )
    var versionSeries: String = ""
) {
    companion object : CEConfig<GameConfig>(
        GameConfig::class,
        "game"
    ) {

        init {
            ConfigRegistry.register(this)
        }

        @JvmStatic
        @JvmOverloads
        fun get(real: Boolean = false): GameConfig = if (real) this.instance() else this.config()
    }

}
